lib/SILOptimizer/Transforms/PerformanceInliner.cpp

```diff
@@ -407,6 +446,12 @@ bool SILPerformanceInliner::isProfitableToInline(
     }
   }
 
+  Weight EntryW = SPA->getWeight(CalleeEntry, CallerWeight);
```

Comment:
Why are you taking the weight of the entry block and not the weight of the block which contains the access(es)?

Reply:
When we visit the block that contains the accesses we might not have visited the entire function / further block might change our decision. by the time we traverse all blocks we 'lost' which block(s) contain accesses. but I agree this is less than ideal - will think about changing it